name: Monthly Test Run

on:
  schedule:
    - cron: "0 0 1 * *"  
  workflow_dispatch:  # Allows manual triggering of tests if needed

jobs:
  test:
    runs-on: ubuntu-latest  

    steps:
      - name: Checkout Repository
        uses: actions/checkout@v3

      - name: Set Up Python
        uses: actions/setup-python@v4
        with:
          python-version: '3.9'  

      - name: Install Dependencies
        run: |
          pip install -r requirements.txt  
          pip install scipy numpy

      - name: Run Unit Tests
        id: test
        run: |
          python -m unittest discover tests  

      - name: Send Email on Failure
        if: failure() 
        uses: dawidd6/action-send-mail@v3
        with:
          server_address: smtp.gmail.com  
          server_port: 587
          username: ${{ secrets.EMAIL_ACCOUNT }}  
          password: ${{ secrets.EMAIL_PASSWORD }}  
          subject: "ðŸ”¥ This Test Suite is a Dumpster Fire ðŸ”¥"
          body: |

            G'day legend, 
            
            Annoyingly, the scheduled test run for the Shine Lab Code has failed. 

            Please check the logs for details, likely an issue with updated modules (e.g. numpy).

            I'd also recommend to double check all results from the last month 
            depending on what the error is.
        
            If you are stuck, reach out to me but then you better get me a coffee:)

            Good luck, 

            Florian Burger

            Repository: ${{ github.repository }}
            Run ID: ${{ github.run_id }}
            Run URL: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
 
          to: fburger20@gmx.de  
          from: "GitHub Actions <your_email@example.com>"

